contracts/PetPark.sol

```diff
@@ -1,7 +1,87 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.0;
 
-
 contract PetPark {
 
+    enum AnimalType {
+        None, 
+        Fish,
+        Cat,
+        Dog,
+        Rabbit,
+        Parrot
+    }
+
+    enum Gender {
+        Male, 
+        Female
+    }
+
+    address public owner;
+    struct Adopter {
+        uint age;
+        Gender gender;
+        AnimalType animalType;
+    }
+
+    mapping(AnimalType => uint) public animalCounts;
+    mapping(address => Adopter) addressAdopters;
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Not an owner");
+        _;
+    }
+
+    modifier validAnimal(AnimalType _animalType) {
+        require(_animalType > AnimalType.None && _animalType <= AnimalType.Parrot, "Invalid animal");
+        _;
+    }
+
+    event Added(AnimalType _animalType, uint _count);
+
+    function add(AnimalType _animalType, uint _count) public onlyOwner validAnimal(_animalType) {
+        animalCounts[_animalType] += _count;
+        emit Added(_animalType, _count);
+    }
+
+    event Borrowed(AnimalType _animalType);
+
+    function borrow(uint _age, Gender _gender, AnimalType _animalType) public {
+        require(_age > 0, "Invalid Age");
+        require(_animalType > AnimalType.None && _animalType <= AnimalType.Parrot, "Invalid animal type");
+
+        Adopter storage existingAdopter = addressAdopters[msg.sender];
```

Comment:
Since `existingAdopter` is only being read from, you can declare it a memory variable which is more gas efficient